src/util/pup.h

```diff
@@ -229,10 +237,16 @@ class er {
   void becomeUserlevel(void) {PUP_er_state|=IS_USERLEVEL;}
   bool isUserlevel(void) const {return (PUP_er_state&IS_USERLEVEL)!=0?true:false;}
   
-  //This indicates that the pup routine is restoring from a checkpoint.
-  void becomeRestarting(void) {PUP_er_state|=IS_RESTARTING;}
-  bool isRestarting(void) const {return (PUP_er_state&IS_RESTARTING)!=0?true:false;}
-  
+  bool isCheckpoint(void) const
```

Comment:
I would think it might be valuable to have separate isRestarting and isCheckpointing, like packing/unpacking. Is isCheckpoint set when restarting from a checkpoint?

Reply:
Yes, isCheckpoint is set both when making a checkpoint and when restarting from a checkpoint. I only found one instance of isRestarting being used in the RTS code, I replaced that with isCheckpoint (in `liveViz`).

Reply:
I like how `isCheckpoint` is orthogonal to which way the data is going. `isPacking && isRestarting` would never happen, and neither would `isUnpacking && isCheckpoint` if the former existed.